test/infrastructure/container/docker.go

```diff
@@ -400,7 +400,7 @@ func (d *dockerRuntime) RunContainer(ctx context.Context, runConfig *RunContaine
 	}
 	networkConfig := network.NetworkingConfig{}
 
-	if runConfig.IPFamily == clusterv1.IPv6IPFamily {
+	if runConfig.IPFamily == clusterv1.IPv6IPFamily || runConfig.IPFamily == clusterv1.DualStackIPFamily {
```

Comment:
Do we have to do something in machine.go func (m *Machine) Address(ctx context.Context) (string, error) { ?

> // Address will get the IP address of the machine. If IPv6 is enabled, it will return
> // the IPv6 address, otherwise an IPv4 address.

This doesn't handle dualstack today. I thik at least the godoc is wrong. I would assume using DualStack means that IPv6 is enabled, but the Address func still returns the IPv4 adress for dualstack

Reply:
Similar in other places. We have a few places where we check for the IPv6 family and if the IPFamily is not set we fallback to IPv4. I wonder if in the DualStack case we should actually use IPv6

(I looked for usages of cluster.IPv6IPFamily across the codebase)